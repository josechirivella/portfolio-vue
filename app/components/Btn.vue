<template>
  <button
    :class="classList"
    class="btn p-4 border-0 rounded-lg"
  >
    <slot />
  </button>
</template>

<script lang="ts" setup>
defineProps({
  classList: { type: String, default: '' },
});
</script>
