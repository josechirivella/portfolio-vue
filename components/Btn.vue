<template>
  <button class="btn p-4 border-0 rounded-lg" :class="props.classList">
    <slot />
  </button>
</template>

<script setup lang="ts">
const props = defineProps({
  classList: { type: String, default: '' },
});
</script>
