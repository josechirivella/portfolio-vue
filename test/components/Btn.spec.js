import { beforeAll, describe, expect, test } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import Btn from '@/components/Btn.vue';

describe('Btn', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallowMount(Btn, {
      props: {
        classList: 'howdy',
      },
    });
  });
  test('should render', () => {
    expect(wrapper.vm).toBeTruthy();
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
  test('should render btn with class', async () => {
    await wrapper.setProps({ classList: 'btn-primary' });
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
