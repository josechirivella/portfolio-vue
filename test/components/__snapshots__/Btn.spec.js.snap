// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Btn > should render 1`] = `
<button
  class="howdy btn p-4 border-0 rounded-lg"
>
  
  
</button>
`;

exports[`Btn > should render btn with class 1`] = `
<button
  class="btn-primary btn p-4 border-0 rounded-lg"
>
  
  
</button>
`;
